src/main/java/net/glowstone/command/minecraft/PlaySoundCommand.java

```diff
@@ -0,0 +1,174 @@
+
+            } catch (final NumberFormatException n) {
+                sender.sendMessage(ChatColor.RED + "'" + args[7] + "' is not a valid number");
+                return false;
+            }
+        }
+
+        if (args.length >= 7) {
+            try {
+                volume = Double.valueOf(args[6]);
+            } catch (final NumberFormatException n) {
+                sender.sendMessage(ChatColor.RED + "'" + args[6] + "' is not a valid number");
+                return false;
+            }
+        }
+
+        if (args.length >= 6) {
+            relativeLocation = args[3].startsWith("~") || args[4].startsWith("~") || args[5].startsWith("~");
+        }
+
+        for (final GlowPlayer target : targets) {
+            Location soundLocation, targetLocation = target.getLocation();
+            double targetVolume = volume;
+
+            try {
+                if (relativeLocation) {
+                    soundLocation = CommandUtils.getLocation(targetLocation, args[3], args[4], args[5]);
+                } else if (args.length >= 6) {
+                    soundLocation = CommandUtils.getLocation(new Location(world, 0, 0, 0), args[3], args[4], args[5]);
+                } else {
+                    soundLocation = targetLocation;
+                }
+            } catch (final NumberFormatException n) {
+                sender.sendMessage(ChatColor.RED + "The position (" + args[3] + ","  + args[4] + "," + args[5] + ") is invalid");
+                return false;
+            }
+
+            // If the target is outside the normal audible sphere
+            if (targetLocation.distanceSquared(soundLocation) > Math.pow(volume, 2)) {
+                if (minimumVolume <= 0) {
+                    sender.sendMessage(ChatColor.RED + target.getName() + " is too far away to hear the sound");
+                    return false;
+                } else {
+                    final double deltaX = soundLocation.getX() - targetLocation.getX(),
+                            deltaY = soundLocation.getX() - targetLocation.getY(),
+                            deltaZ = soundLocation.getX() - targetLocation.getZ();
+                    final double delta = Math.sqrt(Math.pow(deltaX, 2) + Math.pow(deltaY, 2) + Math.pow(deltaZ, 2));
+
+                    soundLocation = targetLocation;
+                    soundLocation.add(deltaX / delta, deltaY / delta, deltaZ / delta);
+                    targetVolume = minimumVolume;
+                }
+            }
+
+            target.playSound(soundLocation, sound, soundCategory, (float) targetVolume, (float) pitch);
+        }
+
+        return true;
+    }
+
+    @Override
+    public List<String> tabComplete(CommandSender sender, String alias, String[] args) throws IllegalArgumentException {
+        if (args == null) {
+            return Collections.emptyList();
+        } else if (args.length == 1) {
+            return StringUtil.copyPartialMatches(args[0], SOUNDS, new ArrayList(SOUNDS.size()));
```

Comment:
You can do something similar to what I did for the [Give command](https://github.com/GlowstoneMC/Glowstone/blob/master/src/main/java/net/glowstone/command/minecraft/GiveCommand.java#L95-L100) (in order to keep the same behavior as vanilla)